### PR TITLE
Update dotnet to v6.0

### LINKED ISSUE #85: Update dotnet version to v6
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

The dotnet image currently has  v3.1 installed, given v6 is now the latest LTS, can the image be updated to use this, or offer a separate v6 version?

### PR DESCRIPTION
**Pull request by Zaid-Ajaj**
From the dotnet docker image. 

Fixes #85 

Part of [pulumi/pulumi#7911](https://github.com/pulumi/pulumi/issues/7911)

Tested by running:
```bash
cd ./docker/dotnet
# build the image
docker build -t pulumi-dotnet --no-cache .
# run interactive bash session within the image
docker run --name pulumi-dotnet-container --rm -i -t pulumi-dotnet bash
# started interactive bash session
# from container > dotnet --version
# from container > 6.0.300
```

### CODE REVIEW

**Comment by Zaid-Ajaj** [comment]
Even when building netcoreapp3.1 projects using dotnet v6.0, it seems now that the runtime fails to connect the pulumi program (client) to pulumi (server). I suspect this is because dotnet v6.0 no longer understands the `AppContext` that is used to [tell gRPC to use insecure connections](https://docs.microsoft.com/en-us/aspnet/core/grpc/troubleshoot?view=aspnetcore-6.0#call-insecure-grpc-services-with-net-core-client)

So running the templates now fails because of gRPC connections closing: doesn't allow insecure connections

Basically blocked by [pulumi/templates#294](https://github.com/pulumi/templates/pull/294)

File: docker/dotnet/Dockerfile [hunk 1]
```diff
@@ -18,17 +18,16 @@ FROM debian:11-slim
 LABEL org.opencontainers.image.description="Pulumi CLI container for dotnet"
 WORKDIR /pulumi/projects
 
-ARG DOTNET_VERSION="3.1"
+ARG DOTNET_VERSION="6.0"
```
Comment (mikhailshilkov) [inline docker/dotnet/Dockerfile:21]: Do you know if this is potentially breaking for users relying on 3.1 in some way?

**Review by mikhailshilkov — APPROVED** [review]
